pkg/multiclusterservice/map.go

```diff
@@ -6,22 +6,25 @@ import (
 	lighthousev1 "github.com/submariner-io/lighthouse/pkg/apis/lighthouse.submariner.io/v1"
 )
 
-type RemoteService struct {
+type serviceInfo struct {
 	key         string
 	ClusterInfo map[string]string
 	IpList      []string
 }
 
 type Map struct {
-	svcMap map[string]*RemoteService
+	svcMap map[string]*serviceInfo
 	sync.RWMutex
 }
 
-func (m *Map) Get(namespace string, name string) (*RemoteService, bool) {
-	return m.GetByKey(keyFunc(namespace, name))
+func (m *Map) GetIps(namespace string, name string) ([]string, bool) {
+	if val, ok := m.getByKey(keyFunc(namespace, name)); ok {
+		return val.IpList, len(val.IpList) > 0
```

Comment:
It's not completely safe to access the `IpList` field in this manner outside the lock - it's possible `IpList` could be re-assigned while evaluating `len`.  I'd suggest to either eliminate `getByKey` or lock here and not in `getByKey`.

Reply:
Argh! I had locked this but Ctrl+Zed it away. Will do.